### PR TITLE
Confluence report accounts workflow

### LINKED ISSUE #3351: [Caloris 2] [Confluence] Implement Personal Data Reporting API
Implement https://developer.atlassian.com/cloud/confluence/user-privacy-developer-guide/

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

Resolves https://github.com/dust-tt/dust/issues/3351.

This PR fulfills the necessary steps to make our Dust App available within the Atlassian community, enabling everyone to link their Confluence (and soon Jira) accounts with Dust. As part of the app-sharing prerequisites, we must adhere to Atlassian's GDPR guidelines (see [documentation](https://developer.atlassian.com/cloud/jira/platform/user-privacy-developer-guide/#user-privacy-guide-for-app-developers)).

The specified endpoint, `/app/report-accounts`, takes a list of user account IDs and their oldest updated times and determines whether we can retain those accounts' data in our infrastructure. If an account is deactivated, we are required to purge all its associated data. This account status check must be conducted every 7 days.

To comply with the 7-day cycle, this PR introduces a Temporal schedule. This schedule will trigger a routine to verify the status of all user accounts. Additionally, we will implement checks in production to guarantee the proper execution of this schedule.

### CODE REVIEW

File: connectors/src/connectors/confluence/temporal/activities.ts [hunk 1]
```diff
@@ -439,3 +439,78 @@ export async function confluenceRemoveSpaceActivity(
...
+  if (!connector) {
+    throw new Error(`Connector not found (connectorId: ${connectorId})`);
+  }
+
+  // We look for the oldest updated data.
+  const oldestPageSync = await ConfluencePage.findOne({
+    where: {
+      connectorId,
+    },
+    order: [["lastVisitedAt", "ASC"]],
```
Comment (spolu) [inline connectors/src/connectors/confluence/temporal/activities.ts:493]: We likely need an index here?

File: connectors/src/connectors/confluence/temporal/client.ts [hunk 2]
```diff
@@ -147,3 +153,37 @@ export async function stopConfluenceSyncWorkflow(
...
+      policies: {
+        // If Temporal Server is down or unavailable at the time when a Schedule should take an Action.
+        // Backfill scheduled action up to 1 previous day.
+        catchupWindow: "1 day",
+        // Only one workflow at a time.
+        overlap: ScheduleOverlapPolicy.SKIP,
+      },
+      spec: {
+        // According to Atlassian's documentation, the cycle period is 7 days.
+        intervals: [{ every: "7d" }],
```
Comment (spolu) [inline connectors/src/connectors/confluence/temporal/client.ts:180]: No need to action here but we'll likely need a production check for all long running confluence workflows?

File: connectors/src/connectors/confluence/temporal/workflows.ts [hunk 3]
```diff
@@ -245,3 +257,33 @@ export async function confluenceRemoveSpaceWorkflow({
...
+      const { memo, searchAttributes: parentSearchAttributes } = workflowInfo();
+
+      // If the account is closed, remove all associated Spaces and Pages from storage.
+      await executeChild(confluenceRemoveSpacesWorkflow, {
+        workflowId: makeConfluenceRemoveSpacesWorkflowId(blob.connectorId),
+        searchAttributes: parentSearchAttributes,
+        args: [
+          {
+            connectorId: blob.connectorId,
+            spaceIds: [],
```
Comment (spolu) [inline connectors/src/connectors/confluence/temporal/workflows.ts:280]: should this be null? a bit more explicit?

File: connectors/src/connectors/confluence/temporal/workflows.ts [hunk 4]
```diff
@@ -245,3 +257,33 @@ export async function confluenceRemoveSpaceWorkflow({
...
+          },
+        ],
+        memo,
+      });
+
+      // TODO(2024-01-23 flav) Implement logic to remove row in the Connector table and stop all workflows.
```
[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/connectors/confluence/temporal/workflows.ts:286]: ack

**Review by spolu — APPROVED** [review]
LGTM

Left a few comments but feel free to merge once addressed :+1:

File: connectors/src/connectors/confluence/temporal/activities.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/src/connectors/confluence/temporal/activities.ts:493]: Yep, I plan to add one in another PR.

File: connectors/src/connectors/confluence/temporal/client.ts (hunk 2 above)
Comment (flvndvd) [reply connectors/src/connectors/confluence/temporal/client.ts:180]: Yes, it's tracked here: https://github.com/dust-tt/dust/issues/3381.

File: connectors/src/connectors/confluence/temporal/workflows.ts (hunk 3 above)
Comment (spolu) [reply connectors/src/connectors/confluence/temporal/workflows.ts:280]: Sure thing

File: connectors/src/connectors/confluence/temporal/activities.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/src/connectors/confluence/temporal/activities.ts:493]: Added in https://github.com/dust-tt/dust/pull/3418.

Comment (flvndvd) [reply connectors/src/connectors/confluence/temporal/activities.ts:493]: ✅

**Pull request merged**